src/utils.jl

```diff
@@ -295,7 +295,8 @@ function lineoffset(framecode::FrameCode)
     return offset
 end
 
-getline(ln) = Int(isexpr(ln, :line) ? ln.args[1] : ln.line)::Int
+getline(ln::LineTypes) = ln.line::Int
+getline(ln::Expr)      = ln.args[1]::Int # assuming ln.head === :line
```

Comment:
The original tries to convert to an `Int` and then asserts it succeeds; this change requires that they start out as `Int`. It also doesn't check that `ln` is a `:line` Expr/ Are you sure these changes are safe?

Reply:
I found the `:line` field of `LineInfoNode` and `LineInfoNode` [are `Int`](https://github.com/JuliaLang/julia/blob/9921e8c1e5934e308f18e748dfd405503bf2b34c/src/jltypes.c#L2089-L2098), but I didn't track how `:line` expression is constructed... 
I think making explicit conversion here is way more robust and doesn't hurt anything, so let me do that again.

---

As for the validation for `:line` expression, I excluded that because I believe we're actually only calling this function on `:line` expressions, and non-`:line` `Expr`s will throw anyway (since it never has `.line` field). We can do something like `getline(ln::Expr) = isexpr(ln, :line) ? ln.args[1] : throw("expected :line expression")`, but I feel it's too much ?